File: back-side/persistence/src/main/java/esp/lord/dic1/uml/persistence/entities/Patient.java
package esp.lord.dic1.uml.persistence.entities;

import esp.lord.dic1.uml.persistence.enums.Sex;
import esp.lord.dic1.uml.persistence.medicalfile.entities.MedicalFile;
import jakarta.persistence.*;
import lombok.*;
import java.util.Date;
import java.util.List;


//[TODO] envoyer à Rawane un template
@NoArgsConstructor @AllArgsConstructor @Setter @Getter @ToString
@Entity
public class Patient {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer id;
    String lastName;
    String firstName;
    @Enumerated(EnumType.STRING)
    Sex sex;
    Date birthDay;
    String email;
    @OneToMany
    List<MedicalFile> medicalFiles;

}

File: front-side/webappts/src/types/patient.ts
export enum Sex {
    MASCULIN,
    FEMININ,
    AUTRE
}

export interface Patient {
    id: number,
    lastName: string,
    firstName: string,
    sex: Sex,
    birthDay: string,
    email: string
}

export interface PatientFields  {
    id?: number,
    lastName: string,
    firstName: string,
    sex: Sex,
    age: number
}
